sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -469,24 +502,16 @@ static rom_error_t rom_ext_attestation_owner(const manifest_t *owner_manifest) {
   if (launder32(cert_valid) == kHardenedBoolFalse) {
     HARDENED_CHECK_EQ(cert_valid, kHardenedBoolFalse);
     dbg_printf("CDI_1 certificate not valid. Updating it ...\r\n");
-    uint32_t updated_cert_size = kCdi1MaxCertSizeBytes;
-    // TODO(#19596): add owner configuration block measurement to CDI_1 cert.
+    size_t cdi_1_cert_size = kCdi1MaxCertSizeBytes;
     HARDENED_RETURN_IF_ERROR(dice_cdi_1_cert_build(
-        (hmac_digest_t *)boot_measurements.bl0.data,
-        (hmac_digest_t *)zero_binding_value.data,
+        (hmac_digest_t *)boot_measurements.bl0.data, &owner_measurement,
         owner_manifest->security_version, &cdi_1_key_ids,
-        &curr_attestation_pubkey, cdi_1_cert, &updated_cert_size));
-    // Update the cert page buffer.
-    memcpy(&dice_certs_page[dice_certs_page_offset], cdi_1_cert,
-           updated_cert_size);
-    dice_certs_page_dirty = kHardenedBoolTrue;
-    rom_ext_attestation_increment_cert_offset(updated_cert_size);
-    // If the CDI_1 cert is updated, it could be smaller than the previous CDI_1
-    // cert, so we make sure to clear out the rest of the buffer.
-    memset(&dice_certs_page[dice_certs_page_offset], UINT8_MAX,
-           FLASH_CTRL_PARAM_BYTES_PER_PAGE - dice_certs_page_offset);
-  } else {
-    rom_ext_attestation_increment_cert_offset(cert_size);
+        &curr_attestation_pubkey, cdi_1_cert, &cdi_1_cert_size));
+    HARDENED_RETURN_IF_ERROR(flash_ctrl_info_erase(&kFlashCtrlInfoPageDiceCerts,
```

Comment:
if we want to update the cert, we need to just write to the buffer in RAM and mark the dirty flag; the buffer will get written back to flash on line 530 if the certs were updated in the buffer.

Reply:
see how it was originally done.

Reply:
Done. PTAL.

Reply:
LGTM. thanks.